packages/graphql/src/translate/cypher-builder/build-merge-statement.ts

```diff
@@ -159,13 +158,16 @@ export function buildMergeStatement({
     return joinStatements([mergeNodeStatement, onCreateSetQuery, ...onCreateStatements]);
 }
 
-function buildOnCreate(onCreate: Record<string, any>, varName: string): CypherStatement {
+function buildOnCreate(onCreate: Record<string, any>, varName: string, node?: Node | undefined): CypherStatement {
     const queries: string[] = [];
     const parameters = {};
 
     Object.entries(onCreate).forEach(([key, value]) => {
-        queries.push(`${varName}.${key} = $${generateParameterKey(`${varName}_on_create`, key)}`);
-        parameters[generateParameterKey(`${varName}_on_create`, key)] = value;
+        const nodeField = node?.primitiveFields.find((f) => f.fieldName === key);
+        const usedFieldName = nodeField ? nodeField.dbPropertyName || nodeField.fieldName : key;
```

Comment:
**suggestion (non-blocking)**: Maybe readability here could be improved (i.e. avoid 2 conditionals in one line)?
```suggestion
        const nodeFieldName = nodeField?.dbPropertyName || nodeField?.fieldName;
        const fieldName = nodeFieldName || key;
```